eng/pipelines/templates/jobs/archetype-sdk-client.yml

```diff
@@ -146,7 +146,7 @@ jobs:
         OSVmImage: ubuntu-20.04
         vcpkg.deps: 'cmocka'
         VCPKG_DEFAULT_TRIPLET: 'x64-linux'
-        AptDependencies: 'gcovr lcov clang-format-9'
+        AptDependencies: 'gcovr lcov clang-format-14'
```

Comment:
Should also include changes to the invocations of `clang-format` here to either `clang-format` or `clang-format-14`: https://github.com/Azure/azure-sdk-for-c/blob/bbe6c1cff6483ebab650a2b6a8087512107cd77f/eng/pipelines/templates/jobs/archetype-sdk-client.yml#L363-L365

Reply:
Indeed `clang-format-14` is not available on Ubuntu 20.04. A couple of options: 

1. Upgrade this matrix entry to Ubuntu 22.04
2. Use a different version of clang-format 

Cursory examination suggests clang-format-12 is available Ubuntu 20.04: 

![image](https://github.com/Azure/azure-sdk-for-c/assets/2158838/e7e37509-7db0-40c2-be35-6615df37fed3)

Reply:
Got it, I will try again using clang-format-12.

Reply:
Gotta use clang-format-12 in the command, not clang-format-9... The script as it was written fails silently. I refactored and made it fail more loudly if the expected command isn't present.

Reply:
When I went to the terminal in the SDK folder, I executed the command with clang-format-12. I did not realize that the .yml file also had to be changed. Now is it resolved or do I commit again with those changes? Thank you.